api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java

```diff
@@ -132,7 +121,7 @@ public void checkFreeTierBillingUsageForUsers(
   private Set<DbUser> filterUsersHigherThanTheLowestThreshold(
       Set<DbUser> users, final Map<Long, Double> liveCostByCreator) {
     final List<Double> costThresholdsInDescOrder =
-        workbenchConfigProvider.get().billing.freeTierCostAlertThresholds;
```

Comment:
cloning the list to avoid modifying it concurrently because we run cloud tasks now which spin multiple threads. We also don't want to sort the config in place anyway

Reply:
We don't actually need to have this list at all, since we only want the minimum value.

Here's my attempt at a streaming equivalent:
```
    return workbenchConfigProvider.get().billing.freeTierCostAlertThresholds.stream()
        .min(Comparator.naturalOrder())
        .map(
            lowestThreshold ->
                users.stream()
                    .filter(
                        user -> {
                          final double limit = getUserFreeTierDollarLimit(user);
                          final double userLiveCost =
                              Optional.ofNullable(liveCostByCreator.get(user.getUserId()))
                                  .orElse(0.0);
                          return userLiveCost / limit >= lowestThreshold;
                        })
                    .collect(Collectors.toSet()))
        .orElse(Collections.emptySet());
```

Reply:
very nice idea!